src/libstore/store-api.hh

```diff
@@ -157,7 +157,9 @@ public:
 
     const Setting<int> pathInfoCacheSize{this, 65536, "path-info-cache-size", "size of the in-memory store path information cache"};
 
-    const Setting<bool> isTrusted{this, false, "trusted", "whether paths from this store can be used as substitutes even when they lack trusted signatures"};
+    const Setting<bool> isTrusted{this, false, "trusted", "whether paths from this store can be used as substitutes even when they lack trusted signatures. Compare \"trusting\""};
+
+    Setting<bool> isTrusting{this, true, "trusting", "whether (we think) paths can be added to this store even when they lack trusted signatures. Compare \"trusted\""};
```

Comment:
We should be careful about adding another signature-related flag since it's getting very hard to understand (we already have `trusted` and `--no-check-sigs`, and it's already not really clear how they interact...).

Why do we actually need "trusting"? Either the remote store accepts the paths we send to it or it doesn't...

Reply:
> We should be careful about adding another signature-related flag since it's getting very hard to understand (we already have `trusted` and `--no-check-sigs`

Agreed, this isn't a new axis, and I am not sure about the differences between `trusted` and `--no-check-sigs` either.

> Either the remote store accepts the paths we send to it or it doesn't...

 Agreed, this isn't a new axis but rather a way to guess whether the remote store would accept non-CA paths or wouldn't. As @matthewbauer says above it would be slicker to not have the user guess but instead try and fallback, but I wasn't sure where to put the state from trying. Maybe I'm overthinking it and it can just be local vars in `copyPaths` and `build-remote`.

Reply:
Using https://github.com/NixOS/nix/pull/7515 we have gotten rid of it!